lib/utils.js

```diff
@@ -748,3 +757,52 @@ exports.stackTraceFilter = function() {
     return stack.join('\n');
   };
 };
+
+//
+/**
+ * Number.MAX_SAFE_INTEGER only available in ES2015.
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+ * @type {number}
+ */
+exports.MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 0x1fffffffffffff;
+
+/**
+ * Attempt to convert `value` an integer, if it's a representation
+ * of a "safe" unsigned integer.
+ * @param {*} [value] Value to convert
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/parseInt
+ * @returns {(number|NaN)} result
+ */
+exports.toSafeUnsignedInteger = function toSafeUnsignedInteger(value) {
+  var intVal;
+  if (/^0[xX]/.test(value)) {
+    intVal = parseInt(value, 16);
+  } else if (/^0[^.]/.test(value)) {
+    intVal = parseInt(value, 8);
+  } else {
+    intVal = parseInt(value, 10);
+  }
+  return !isNaN(intVal) && parseFloat(intVal) === intVal
+    && exports.isSafeUnsignedInteger(intVal)
+    ? intVal
+    : NaN;
+};
+
+/**
+ * Given integer `int`, try to convert it to a hex string.
+ * @param {number} int Integer
+ * @returns {string} Hexadecimal representation thereof
+ */
+exports.toHex = function toHex(int) {
+  return '0x' + int.toString(16).toUpperCase();
+};
+
+/**
+ * Returns `true` if `value` is a "safe" 32-bit unsigned int.
```

Comment:
Here and in various other places, it's a bit confusing to see uint32_t or integers in the range of [0, 2³² – 1] being called "safe," while what the function actually takes is integers in the range of [1, 2⁵³ – 1]. IMO, what we should allow is [0, 2⁵³ – 1] which can be described as "safe" unsigned 53-bit integer and matches with ECMAScript semantics. (In addition, we can optionally allow negative values as well.)